lib/util/page-number-functions.js

```diff
@@ -0,0 +1,71 @@
+const { removeLeadingWhitespaces, removeTrailingWhitespaces, isNumber } = require('./string-functions')
+
+const searchRange = (numerator, denominator, length) => {
+  return Math.floor(numerator / denominator * length)
+}
+
+const searchArea = (start, end, pageIndexNumMap, pageIndex, items) => {
+  for (let x = start; x < end; x++) {
+    const trimLeadingWhitespaces = removeLeadingWhitespaces(items[x].str)
+    const trimWhitespaces = removeTrailingWhitespaces(trimLeadingWhitespaces)
+    if (isNumber(trimWhitespaces)) {
+      if (!pageIndexNumMap[pageIndex]) {
+        pageIndexNumMap[pageIndex] = []
+      }
+      pageIndexNumMap[pageIndex].push(Number(trimWhitespaces))
+    }
+  }
+  return pageIndexNumMap
+}
+
+exports.findPageNumbers = (pageIndexNumMap, pageIndex, items) => {
+  const topArea = searchRange(1, 6, items.length)
+  const bottomArea = searchRange(5, 6, items.length)
+
+  const topAreaResult = searchArea(0, topArea, pageIndexNumMap, pageIndex, items)
```

Comment:
I think the implementation for `searchArea()` can be simplified by taking `items.slice(start, end)` and feeding that instead. the for loop above can then become...
```js
const searchArea = (...) {
  for (const { str } of items) {
    // ...
```